gptel.el

```diff
@@ -2794,24 +2816,48 @@ for tool call results.  INFO contains the state of the request."
       (when gptel-include-tool-results
         (with-current-buffer (marker-buffer start-marker)
           (cl-loop
-           for (name args result) in response
+           for (tool-call result) in response
```

Comment:
This changes the tool-use API, and also makes the check at [line 2732](https://github.com/karthink/gptel/pull/626/files#diff-58ae8200f2acac302967edbab7a33d6a0a6832a2b6dae6868375f4e5d1968795R2732) fail... I think?  Need to take a closer look.